apps/docs/pages/guides/auth/server-side/creating-a-client.mdx

```diff
@@ -0,0 +1,420 @@
+import Cookies from 'cookies'
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  const cookies = new Cookies(context.req, context.res)
+
+  const supabase = createServerClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+    {
+      cookies: {
+        get(key: string) {
+          return cookies.get(key)
+        },
+        set(key: string, value: string, options: CookieOptions) {
+          cookies.set(key, value, options)
+        },
+        remove(key: string, options: CookieOptions) {
+          cookies.set(key, '', options)
+        },
+      },
+    }
+  )
+
+  return {...}
+}
+```
+
+</TabPanel>
+
+<TabPanel id="get-static-props" label="getStaticProps">
+
+The `getStaticProps` function gets executed at build time, rather than per request. At build time,
+there is no user, session or cookies. Therefore, we can use the standard `createClient` function
+from `supabase-js`.
+
+```ts page.tsx
+import { createClient } from '@supabase/supabase-js'
+
+export const getStaticProps = async () => {
+  const cookies = new Cookies(context.req, context.res)
+
+  const supabase = createClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+  )
+
+  return {...}
+}
+```
+
+</TabPanel>
+
+<TabPanel id="api-route" label="API Route">
+
+Cookies have many edge cases when not using a helper function, therefore, we recommend using the `cookies` package in API Routes to provide a similar API to the Next.js `cookies` function in the App Router.
+
+```bash
+npm install cookies
+```
+
+Types can optionally be installed as well.
+
+```bash
+npm install --save-dev @types/cookies
+```
+
+```ts index.tsx
+import { createServerClient, type CookieOptions } from "@supabase/ssr"
+import { type NextApiRequest, type NextApiResponse } from "next"
+import Cookies from "cookies"
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const cookies = new Cookies(req, res)
+
+  const supabase = createServerClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+    {
+      cookies: {
+        get(key: string) {
+          return cookies.get(key)
+        },
+        set(key: string, value: string, options: CookieOptions) {
+          cookies.set(key, value, options)
+        },
+        remove(key: string, options: CookieOptions) {
+          cookies.set(key, "", options)
+        },
+      },
+    }
+  )
+
+  res.send(...)
+}
+```
+
+</TabPanel>
+
+<TabPanel id="pages-component" label="Pages Component">
+
+```ts index.tsx
+export default function Index () {
+  const supabase = createBrowserClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!
+  )
+
+  return ...
+}
+```
+
+</TabPanel>
+</Tabs>
+
+</TabPanel>
+<TabPanel id="sveltekit" label="SvelteKit">
+
+HELP ME SUPADREW!
```

Comment:
@silentworks could you add examples for creating a browser client and creating a server client for SvelteKit?